File: ExcelDNAVSExtension/CSProjectTemplate/MyRibbon.cs
using System.Runtime.InteropServices;
using ExcelDna.Integration.CustomUI;

namespace $safeprojectname$
{
    [ComVisible(true)]
public class MyRibbon : ExcelRibbon
{
    public override string GetCustomUI(string RibbonID)
    {
        using (System.IO.Stream stream = System.Reflection.Assembly.GetExecutingAssembly().GetManifestResourceStream(
            "$safeprojectname$." + "MyRibbon.xml"))
        {
            using (System.IO.StreamReader reader = new System.IO.StreamReader(stream))
            {
                return reader.ReadToEnd();
            }
        }
    }

    public void OnButtonPressed(IRibbonControl control)
    {
        System.Windows.Forms.MessageBox.Show("Hello!");
    }
}
}

File: ExcelDNAVSExtension/VBProjectTemplate/MyRibbon.vb
Imports System.IO
Imports System.Runtime.InteropServices
Imports ExcelDna.Integration.CustomUI

<ComVisible(True)>
Public Class MyRibbon
    Inherits ExcelRibbon

    Public Overrides Function GetCustomUI(RibbonID As String) As String
        Using stream As Stream = System.Reflection.Assembly.GetExecutingAssembly().GetManifestResourceStream("$safeprojectname$.MyRibbon.xml")
            Using reader As New StreamReader(stream)
                Return reader.ReadToEnd()
            End Using
        End Using
    End Function

    Public Sub OnButtonPressed(control As IRibbonControl)
        System.Windows.Forms.MessageBox.Show("Hello!")
    End Sub

End Class